pkg/services/folder/folderimpl/folder.go

```diff
@@ -508,10 +520,14 @@ func (s *Service) Move(ctx context.Context, cmd *folder.MoveFolderCommand) (*fol
 	})
 }
 
-func (s *Service) nestedFolderDelete(ctx context.Context, cmd *folder.DeleteFolderCommand) error {
+// nestedFolderDelete inspects the folder referenced by the cmd argument, deletes all the entries for
+// its descendant folders (folders which are nested within it either directly or indirectly) from
+// the folder store and returns the UIDs for all its descendants.
```

Comment:
As far as I can tell, because `nestedFolderDelete` is recursive, it returns the UIDs of all its descendants, _both direct and indirect_. We're going progressively deeper into the nested folders and appending the UIDs of all the subfolders we find: 
https://github.com/grafana/grafana/blob/9a023c347cab8a374ff8ccdca44d7f73bd5e03ab/pkg/services/folder/folderimpl/folder.go#L546-L554
Please let me know if I'm missing something though.

Reply:
you are right, I had in mind only the first iteration